de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaDataSubGenerator.xtend

```diff
@@ -0,0 +1,197 @@
+     */
+    val myDataModel = DATA_FACTORY.createDataModel
+    val myVersion = DATA_FACTORY.createVersion
+    val myContext = DATA_FACTORY.createContext
+    /** Seperator is used to build the full qualified names during the generation */
+    val seperator = "."
+    /** OpenAPI schema which will be used as source for generation */
+    val OpenAPI openAPI
+    /** SLF4j Logger */
+    val static logger = LoggerFactory.getLogger(LemmaDataSubGenerator)
+    /** Location where the generated file is written */
+    val String targetFolder
+
+    new(OpenAPI openAPI, String genPath, String dataFilename) {
+        super()
+        logger.debug("Creating new Data Sub Generator...")
+        this.openAPI = openAPI
+        this.targetFolder = genPath + dataFilename
+    }
+
+    def DataModel generate() {
+        logger.debug("Starting generation...")
+        logger.debug("Initializing model instance...")
+        initilize()
+        logger.debug("...data model initialized!")
+        logger.debug("Creating data structures...")
+
+        openAPI?.components?.schemas?.forEach[
+            key, value| getOrCreateDataStructure(myContext, key, value)
+        ]
+        logger.debug("...data structures created!")
+
+        if (OpenApiUtil.writeModel(myDataModel, targetFolder)) {
+            logger.info("Data model generation successful!")
+            logger.info('''Model written to «targetFolder»''')
+        } else
+            throw new Exception("Data model generation failed :(")
+        return myDataModel
+    }
+
+    def initilize() {
+        // Data Model
+        myDataModel.versions.add(myVersion)
+        // Version
+        myVersion.name = OpenApiUtil.removeInvalidCharsFromName(openAPI.info.version)
+        myVersion.contexts.add(myContext)
+        // Context
+        myContext.name = OpenApiUtil.removeInvalidCharsFromName(openAPI.info.title)
+        myContext.version = myVersion
+    }
+
+    def DataStructure getOrCreateDataStructure(
+        Context c, String name, Schema schema
+    ) {
+        var foundObject = findDataStructure(c, StringUtils.capitalize(name))
+        if (foundObject === null) {
+            val newDataStructure = DATA_FACTORY.createDataStructure
+            newDataStructure.name = StringUtils.capitalize(name)
+            c.complexTypes.add(newDataStructure)
+            addDataFieldsToDataStructure(newDataStructure, name, schema)
+            createdDataStructures.put(
+                newDataStructure.buildQualifiedName(seperator), newDataStructure
+            )
+            return newDataStructure
+        } else {
+            return foundObject
+        }
+    }
+
+    def ListType getOrCreateListStructure(Context c, String name, Schema schema) {
+        var foundObject = findListStructure(c, name)
+        if (foundObject === null) {
+            val newListStructure = DATA_FACTORY.createListType
+            newListStructure.name = '''«StringUtils.capitalize(name)»List'''
```

Comment:
Phew... for such simple templates use the "Java concatentation form" to improve comprehensibility:
```
StringUtils.capitalize(name) + "List"
```